Patternlab/dp 14006  update manage your account

clair0917 opened the pull request:
## Description
Update "Manage your account" stickyNav id

## Related Issue / Ticket

- [DP-14006](https://jira.mass.gov/browse/DP-14006)

## Steps to Test
<!-- Outline the steps to test or reproduce the PR here.  Whenever possible deploy your branch to your fork Github Pages so UAT can be done without rebuilding. See: https://github.com/massgov/mayflower/blob/master/docs/deploy.md -->

1. from the /?p=organisms-account-manager pattern check that the H3 id="manage-your-account"
https://mayflower.digital.mass.gov/b/patternlab/DP-14006--update-manage-your-account-id/patterns/03-organisms-by-author-account-manager/03-organisms-by-author-account-manager.html

## Screenshots
![Screen Shot 2019-05-20 at 10 10 23 AM](https://user-images.githubusercontent.com/18662734/58035567-ac1e9000-7aee-11e9-8942-efdc72e60639.png)


## Note:
This PR fixes a sticky nav issue encountered while Melissa was working on [DP-10836](https://jira.mass.gov/browse/DP-10836)

clair0917 committed:
DP-14006: update h3 #

clair0917 committed:
DP-14006: add changelog

clair0917 committed:
DP-14006: re-running circleCI test

ygannett commented on patternlab/styleguide/source/_patterns/03-organisms/by-author/account-manager.twig:
```diff
@@ -1,6 +1,6 @@
 <section class="ma__account-manager">
   {# Static title, users can not change #}
-  <h3 id="manage-account" class="ma__comp-heading">Manage your account</h3>
+  <h3 id="manage-your-account" class="ma__comp-heading">Manage your account</h3>
```
To make the heading level flexible depending on where it's placed in a page in case the template is used for other purposes, can you add the comp heading with include? So, the heading can have a proper heading level number based on where it's placed.

sample:  patternlab/styleguide/source/_patterns/03-organisms/by-author/contact-list.twig

```
  {% set contactHeading = 2 %}

  {% if contactList.compHeading %}
    {% set compHeading = contactList.compHeading %}
    {% include "@atoms/04-headings/comp-heading.twig" %}
    {% set contactHeading = (compHeading.level ? : contactHeading) + 1 %}
  {% endif %}
```
You can pass `id` value to `@atoms/04-headings/comp-heading.twig` according to `@atoms/04-headings/comp-heading.json`:
```
 {
  "compHeading": {
    "title": "Title text",
    "titleContext": "",
    "level": "",
    "color": "",
    "id": "GUID9827924",
    "centered": ""
  }
}
```

ygannett reviewed (changes requested):
Please see below comment about the heading setup.

clair0917 committed:
DP-14006: switch link templates

clair0917 committed:
DP-14006: switch wrapper templates

clair0917 committed:
DP-14006: update readme

clair0917 committed:
DP-14006: add .md file for links

Commit:
DP-14006: Update descriptive-link-wrapper

Updated new descriptive-links-wrapper to ensure link data is passed
on to new descriptive-link molecule.

merauluka commented:
We've updated the patterns available here to make them a bit more abstract so they could be used  in more than one place. @clair0917 rightly noted that a "descriptive link" template would be valuable in the system. I provided a small update to how link data is passed through to the new "descriptive link" molecule which should resolve issues with json demo data note loading.

Commit:
DP-14006: Update changelog to reflect newly generated patterns

clair0917 committed:
DP-14006: update css for descriptive links

clair0917 committed:
DP-14006: update howto page template

clair0917 committed:
DP14006: backstop file udpates

ygannett reviewed (approved):
LGTM.
